### ISSUE TITLE
Possible issue: per_token_logps - per_token_logps.detach()

### ISSUE THREAD

**Issue by odats** [lsdefine/simple_GRPO#33]
In your file grpo_vllm_one.py there is a line: 
`per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages`

`per_token_logps - per_token_logps.detach()` always 0.

**Comment by odats** [comment]
It is not an issue. After reading more about GRPO I understand why it works.
